pkg/operator/certrotation/signer_test.go

```diff
@@ -19,6 +23,244 @@ import (
+	}
+	// not-after and not-before annotations are filled when new signer is generated
+	if err := setSigningCertKeyPairSecret(existing, 24*time.Hour); err != nil {
+		t.Fatal(err)
+	}
+
+	// give it a second so we have a unique signer name,
+	// and also unique not-after, and not-before values
+	<-time.After(2 * time.Second)
+
+	// get the original crt and key bytes to compare later
+	tlsCertWant, ok := existing.Data["tls.crt"]
+	if !ok || len(tlsCertWant) == 0 {
+		t.Fatalf("missing data in 'tls.crt' key of Data: %#v", existing.Data)
+	}
+	tlsKeyWant, ok := existing.Data["tls.key"]
+	if !ok || len(tlsKeyWant) == 0 {
+		t.Fatalf("missing data in 'tls.key' key of Data: %#v", existing.Data)
+	}
+
+	// copy the existing object before test begins, so we can diff it against
+	// the final object on the cluster after the controllers finish
+	secretWant := existing.DeepCopy()
+	indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	syncCacheFn := func(t *testing.T, obj runtime.Object) {
+		switch {
+		case obj == nil:
+			if err := indexer.Delete(existing); err != nil {
+				t.Fatalf("unexpected error while syncing the cache, op=delete: %v", err)
+			}
+		default:
+			indexer.Delete(obj)
+			if err := indexer.Add(obj); err != nil {
+				t.Fatalf("unexpected error while syncing the cache: %v", err)
+			}
+		}
+	}
+	syncCacheFn(t, existing)
+	clientset := kubefake.NewSimpleClientset(existing)
+
+	// the list cache is synced as soon as we have a delete, create, or update
+	clientset.PrependReactor("delete", "secrets", func(action clienttesting.Action) (handled bool, ret runtime.Object, err error) {
+		syncCacheFn(t, nil)
+		return false, nil, nil
+	})
+	clientset.PrependReactor("create", "secrets", func(action clienttesting.Action) (handled bool, ret runtime.Object, err error) {
+		switch action := action.(type) {
+		case clienttesting.CreateActionImpl:
+			syncCacheFn(t, action.GetObject())
+			return false, action.GetObject(), nil
+		}
+		t.Fatalf("wrong test setup, expected an action object of %T", clienttesting.CreateActionImpl{})
+		return false, nil, nil
+	})
+	clientset.PrependReactor("update", "secrets", func(action clienttesting.Action) (handled bool, ret runtime.Object, err error) {
+		switch action := action.(type) {
+		case clienttesting.UpdateActionImpl:
+			syncCacheFn(t, action.GetObject())
+			return false, action.GetObject(), nil
+		}
+		t.Fatalf("wrong test setup, expected an action object of %T", clienttesting.UpdateActionImpl{})
+		return false, nil, nil
+	})
+
+	options := events.RecommendedClusterSingletonCorrelatorOptions()
+	client := clientset.CoreV1().Secrets(ns)
+	newControllerFn := func(ctrlName string, wrapped *wrapped) *RotatedSigningCASecret {
+		recorder := events.NewKubeRecorderWithOptions(clientset.CoreV1().Events(ns), options, "operator", &corev1.ObjectReference{Name: ctrlName, Namespace: ns})
+		return &RotatedSigningCASecret{
```

Comment:
I think that we should also create a test for the other controller (`RotatedSelfSignedCertKeySecret`) , wdyt ?